pyramid_swagger/tween.py

```diff
@@ -93,6 +99,28 @@ def _get_validation_context(registry):
         return noop_context
 
 
+def get_swagger_handler(settings, route_info, registry):
+    """This returns back `settings.swagger_handler` in almost of the cases.
+
+    Exception is made when `settings.include_2dot0_routes` are non-empty and
+    ['1.2', '2.0'] both are present in available swagger versions. In this
+    special scenario, '2.0' spec is chosen only for requests which are listed
+    in the `include_2dot0_routes`. This helps in incremental migration of
+    routes from v1.2 to v2.0 by making moving to v2.0 opt-in.
+
+    :rtype: :class:`SwaggerHandler`
+    """
+    enabled_swagger_versions = get_swagger_versions(registry.settings)
+    if (SWAGGER_20 in enabled_swagger_versions
```

Comment:
these conditions may need some explanations. This special condition is checked if both v1.2 and v2.0 versions are enabled. If no `include_2dot0_routes` is passed, then this special case should not be checked (this will be the usual flow). If `include..routes` is non-empty then check if given route is not present in the included routes, if they are not, use the v1.2 flow for this request.